examples/src/main/java/com/mapbox/navigation/examples/core/ReplayEngineActivity.kt

```diff
@@ -0,0 +1,266 @@
+                                    // .useDefaultLocationEngine(true)
+                                    .build()
+
+                    locationComponent.activateLocationComponent(locationComponentActivationOptions)
+                    locationComponent.isLocationComponentEnabled = true
+                    locationComponent.cameraMode = CameraMode.TRACKING
+                    locationComponent.renderMode = RenderMode.COMPASS
+
+                    initLocationEngine()
+                }
+            }
+            false -> {
+                permissionsManager =
+                        PermissionsManager(this).also { it.requestLocationPermissions(this) }
+            }
+        }
+    }
+
+    fun initLocationEngine() {
+        val requestLocationUpdateRequest =
+                LocationEngineRequest.Builder(DEFAULT_INTERVAL_IN_MILLISECONDS)
+                        .setPriority(LocationEngineRequest.PRIORITY_NO_POWER)
+                        .setMaxWaitTime(DEFAULT_MAX_WAIT_TIME)
+                        .build()
+
+        locationEngine?.requestLocationUpdates(
+                requestLocationUpdateRequest,
+                locationListenerCallback,
+                mainLooper
+        )
+        locationEngine?.getLastLocation(locationListenerCallback)
+    }
+
+    private val routesReqCallback = object : RoutesRequestCallback {
+        override fun onRoutesReady(routes: List<DirectionsRoute>) {
+            Timber.d("route request success %s", routes.toString())
+            if (routes.isNotEmpty()) {
+                navigationMapboxMap?.drawRoute(routes[0])
+                replayRouteLocationEngine.assign(routes[0])
+                startNavigation.visibility = View.VISIBLE
+            } else {
+                startNavigation.visibility = View.GONE
+            }
+        }
+
+        override fun onRoutesRequestFailure(throwable: Throwable, routeOptions: RouteOptions) {
+            Timber.e("route request failure %s", throwable.toString())
+        }
+
+        override fun onRoutesRequestCanceled(routeOptions: RouteOptions) {
+            Timber.d("route request canceled")
+        }
+    }
+
+    @SuppressLint("MissingPermission")
+    fun initListeners() {
+        startNavigation.setOnClickListener {
+            if (mapboxNavigation?.getRoutes()?.isNotEmpty() == true) {
+                navigationMapboxMap?.updateLocationLayerRenderMode(RenderMode.GPS)
+                navigationMapboxMap?.updateCameraTrackingMode(NavigationCamera.NAVIGATION_TRACKING_MODE_GPS)
+                navigationMapboxMap?.startCamera(mapboxNavigation?.getRoutes()!![0])
+            }
+            mapboxNavigation?.startTripSession()
+            startNavigation.visibility = View.GONE
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        mapView.onStart()
+        mapboxNavigation?.registerLocationObserver(locationObserver)
```

Comment:
Should we do the `register` and `unregister` inside `onCreate` and `onDestroy` respectively? 🤔